diff --git a/node_modules/react-native-router-flux/src/Router.js b/node_modules/react-native-router-flux/src/Router.js
index 27c3402..13e1bc1 100644
--- a/node_modules/react-native-router-flux/src/Router.js
+++ b/node_modules/react-native-router-flux/src/Router.js
@@ -18,6 +18,8 @@ class App extends React.Component {
     uriPrefix: null,
     onDeepLink: null,
   };
+  linkingSubscription = null;
+
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.props.backAndroidHandler || this.onBackPress);
@@ -25,12 +27,16 @@ class App extends React.Component {
     // If the app was "woken up" by an external route.
     Linking.getInitialURL().then(url => this.parseDeepURL(url));
     // Add an event listener for further deep linking.
-    Linking.addEventListener('url', this.handleDeepURL);
+    // Linking.addEventListener('url', this.handleDeepURL);
+    this.linkingSubscription = Linking.addEventListener('url', this.handleDeepURL);
   }
 
   componentWillUnmount() {
     BackHandler.removeEventListener('hardwareBackPress', this.props.backAndroidHandler || this.onBackPress);
-    Linking.removeEventListener('url', this.handleDeepURL);
+    // Linking.removeEventListener('url', this.handleDeepURL);
+    if (this.linkingSubscription) {
+      this.linkingSubscription.remove();
+    }
   }
 
   onBackPress = () => navigationStore.pop();
