diff --git a/node_modules/react-native-confirmation-code-input/components/ConfirmationCodeInput.js b/node_modules/react-native-confirmation-code-input/components/ConfirmationCodeInput.js
index 239c85a..1633330 100644
--- a/node_modules/react-native-confirmation-code-input/components/ConfirmationCodeInput.js
+++ b/node_modules/react-native-confirmation-code-input/components/ConfirmationCodeInput.js
@@ -69,11 +69,11 @@ export default class ConfirmationCodeInput extends Component {
   }
   
   _setFocus(index) {
-    this.codeInputRefs[index].focus();
+    this.codeInputRefs[index]?.focus();
   }
   
   _blur(index) {
-    this.codeInputRefs[index].blur();
+    this.codeInputRefs[index]?.blur();
   }
   
   _onFocus(index) {
@@ -195,9 +195,15 @@ export default class ConfirmationCodeInput extends Component {
   
   _onKeyPress(e) {
     if (e.nativeEvent.key === 'Backspace') {
+      // Return if duration between previous key press and backspace is less than 20ms
+      if (Math.abs(this.lastKeyEventTimestamp - e.timeStamp) < 20) return;
+
       const { currentIndex } = this.state;
       const nextIndex = currentIndex > 0 ? currentIndex - 1 : 0;
       this._setFocus(nextIndex);
+    } else {
+      // Record non-backspace key event time stamp
+      this.lastKeyEventTimestamp = e.timeStamp;
     }
   }
   
